### DISCUSSION TITLE
[Share] 删除摘要中的多余空格

### DISCUSSION THREAD

**Discussion by redleafnew** [windingwind/zotero-actions-tags#196] [Action Scripts]
### Description

需求来源于：https://github.com/redleafnew/zotero-updateifsE/issues/124

### Event

None

### Operation

Script

### Data

```javascript
/**
 * A description of this script.
 * @author Zotero Community
 * @usage 
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/
 */

var pattern = /\s{2,}/g;
var pattern1 = /\s(\d)/g;
var pattern2 = /(\d)\s(\/)/g;

if (!item) return;

if (item.isRegularItem() && !item.isCollection()) {

    var newAb = item.getField('abstractNote').replace(pattern, ' ')  // 多个空格替换为一个空格
        .replace(pattern1, "$1") //替换数字后面的空格
	.replace(pattern2, "$1$2");  // 清除数字和/之间的空格
    item.setField('abstractNote', newAb);  //替换摘要

    return "摘要中的多余空格已删除！"
}
```

**Comment by zhangshd** [comment]
请问这个脚本该怎么用啊，往哪里放呢？、

**Comment by redleafnew** [comment]
https://github.com/windingwind/zotero-actions-tags#readme

**Reply by zhangshd** [reply]
![image](https://github.com/windingwind/zotero-actions-tags/assets/23273686/780e6442-8bea-468b-8111-74dd13560921)
我用的zotero6只有tag功能，没有action功能，好像没法加进去

**Reply by windingwind** [reply]
仅支持Z7版本。

**Comment by redleafnew** [comment]
@windingwind 有办法吗？

**Comment by redleafnew** [comment]
那如果不想升级z7，就用这个吧
https://github.com/redleafnew/zotero-javascripts/blob/main/27remove-extra-space-in-abstract.js

**Reply by zhangshd** [reply]
这个要怎么用呢？（抱歉刚开始上手zotero啥都不懂
![26929A12](https://github.com/windingwind/zotero-actions-tags/assets/23273686/b47ab76e-b78e-43a7-b8b2-0702f27c8256)
）

**Reply by redleafnew** [reply]
https://github.com/redleafnew/zotero-javascripts/tree/main#readme

**Reply by zhangshd** [reply]
万分感谢，您太有耐心了！！

**Reply by nkh0472** [reply]
最近几个月似乎每次触发的时候会报错item.isCollection()函数不存在还是什么的，提示框一闪而过

**Comment by mjthoraval** [comment]
What is the difference between this script and the one provided by  @windingwind here: https://github.com/windingwind/zotero-actions-tags/discussions/163?
Or with the Abstract Cleaner plugin for Zotero 6: https://github.com/dcartertod/zotero-plugins?

**Reply by redleafnew** [reply]
it is suitable to https://github.com/redleafnew/zotero-updateifsE/issues/124

**Reply by redleafnew** [reply]
This one only removes extra spaces.

**Reply by mjthoraval** [reply]
I am using the script from the Abstract Cleaner plugin to remove extra line endings in some of the abstracts.
So I am interested in understanding if I should also use your script separately to do something different (if they have different functionalities and purposes), if the two scripts can be consolidated to do the job of abstract cleaning even better, or if they are basically doing the same with a few subbtle differences?

After reading through the issue that you link to, the problem in there seems to be exactly the same as the problem that is addressed by the script from the Abstract Cleaner plugin. This is why I am curious to understand what is different in your code.
Thank you very much!

**Comment by LXSKING** [comment]
<img width="275" height="76" alt="image" src="https://github.com/user-attachments/assets/478080ff-d86d-49ce-9bea-fe6e4de9a008" />
我这里还是提示有问题。
版本号 7.1-beta.23+6787dcd59

**Reply by LXSKING** [reply]
/**
 * Clean abstract spaces
 */

var pattern = /\s{2,}/g;
var pattern1 = /\s(\d)/g;
var pattern2 = /(\d)\s(\/)/g;

if (!item) return;

// 只处理普通文献条目
if (item.isRegularItem()) {

    var abstract = item.getField('abstractNote');
    if (abstract) {
        var newAb = abstract
            .replace(pattern, ' ')  // 多个空格替换为一个空格
            .replace(pattern1, "$1") // 替换数字前的空格
            .replace(pattern2, "$1$2");  // 清除数字和/之间的空格

        item.setField('abstractNote', newAb);  // 替换摘要
        return "摘要中的多余空格已删除！";
    } else {
        return "摘要为空，无需处理。";
    }
}
我用GPT改了下，好像可以用了。
最后会合成一个段落。
